e2e/testcases/helm_sync_test.go

```diff
@@ -879,3 +913,148 @@ func helmChartVersion(chartVersion string) nomostest.Sha1Func {
 		return chartVersion, nil
 	}
 }
+
+// fetchServiceAccountKeyFile downloads a service account key from google
+// secret manager, using the specified `gsaKeySecretID`. If the key has expired,
+// a new key is generated and stored as a new version of the secret in secret
+// manager. If successful, the old expired key is deleted. Then the path to the
+// new key file is returned. The key file will be deleted when the current
+// test ends.
+func fetchServiceAccountKeyFile(nt *nomostest.NT, projectID, gsaKeySecretID, gsaEmail, gsaName string) (string, error) {
+	// Use a temp file to hold the secret to avoid logging the secret contents
+	gsaKeyFilePath := filepath.Join(nt.TmpDir, fmt.Sprintf("%s.json", gsaName))
+	nt.T.Cleanup(func() {
+		nt.Must(os.RemoveAll(gsaKeyFilePath))
+	})
+
+	nt.T.Log("Listing enabled service account key versions in Secret Manager")
+	out, err := nt.Shell.ExecWithDebug("gcloud", "secrets", "versions", "list",
+		gsaKeySecretID,
+		"--filter", "state=ENABLED",
+		"--limit", "1",
+		"--format", "value(name)",
+		"--project", projectID)
+	if err != nil {
+		// TODO: create new key and secret if no secret found
```

Comment:
Might be worth doing this now to avoid a gotcha when bootstrapping new environments